pkg/health/server/server.go

```diff
@@ -170,9 +174,114 @@ func (s *Server) updateCluster(report *healthReport) {
+		} else if targetNodeName != localNodeName {
+			location = metrics.LabelLocationRemoteIntraCluster
+		}
+
+		// Aggregated status for endpoint connectivity
+		metrics.NodeConnectivityStatus.WithLabelValues(
+			localClusterName, localNodeName, targetClusterName, targetNodeName, location, metrics.LabelPeerEndpoint).
+			Set(metrics.BoolToFloat64(isEndpointReachable))
+
+		// Aggregated status for node connectivity
+		metrics.NodeConnectivityStatus.WithLabelValues(
+			localClusterName, localNodeName, targetClusterName, targetNodeName, location, metrics.LabelPeerNode).
+			Set(metrics.BoolToFloat64(isNodeReachable))
+
+		// HTTP endpoint primary
+		collectConnectivityMetric(endpointPathStatus.PrimaryAddress.HTTP, localClusterName, localNodeName,
+			targetClusterName, targetNodeName, endpointPathStatus.PrimaryAddress.IP,
+			location, metrics.LabelPeerEndpoint, metrics.LabelTrafficHTTP, metrics.LabelAddressTypePrimary)
+
+		// HTTP endpoint secondary
+		for _, secondary := range endpointPathStatus.SecondaryAddresses {
+			collectConnectivityMetric(secondary.HTTP, localClusterName, localNodeName,
+				targetClusterName, targetNodeName, endpointPathStatus.PrimaryAddress.IP,
+				location, metrics.LabelPeerEndpoint, metrics.LabelTrafficHTTP, metrics.LabelAddressTypeSecondary)
+		}
+
+		// HTTP node primary
+		collectConnectivityMetric(nodePathPrimaryAddress.HTTP, localClusterName, localNodeName,
+			targetClusterName, targetNodeName, nodePathPrimaryAddress.IP,
+			location, metrics.LabelPeerNode, metrics.LabelTrafficHTTP, metrics.LabelAddressTypePrimary)
+
+		// HTTP node secondary
+		for _, secondary := range nodePathSecondaryAddress {
+			collectConnectivityMetric(secondary.HTTP, localClusterName, localNodeName,
+				targetClusterName, targetNodeName, secondary.IP,
+				location, metrics.LabelPeerNode, metrics.LabelTrafficHTTP, metrics.LabelAddressTypeSecondary)
+		}
+
+		// ICMP endpoint primary
+		collectConnectivityMetric(endpointPathStatus.PrimaryAddress.Icmp, localClusterName, localNodeName,
+			targetClusterName, targetNodeName, endpointPathStatus.PrimaryAddress.IP,
+			location, metrics.LabelPeerEndpoint, metrics.LabelTrafficICMP, metrics.LabelAddressTypePrimary)
+
+		// ICMP endpoint secondary
+		for _, secondary := range endpointPathStatus.SecondaryAddresses {
+			collectConnectivityMetric(secondary.Icmp, localClusterName, localNodeName,
+				targetClusterName, targetNodeName, secondary.IP,
+				location, metrics.LabelPeerEndpoint, metrics.LabelTrafficICMP, metrics.LabelAddressTypeSecondary)
+		}
+
+		// ICMP node primary
+		collectConnectivityMetric(nodePathPrimaryAddress.HTTP, localClusterName, localNodeName,
+			targetClusterName, targetNodeName, nodePathPrimaryAddress.IP,
+			location, metrics.LabelPeerNode, metrics.LabelTrafficICMP, metrics.LabelAddressTypePrimary)
+
+		// ICMP node secondary
+		for _, secondary := range nodePathSecondaryAddress {
+			collectConnectivityMetric(secondary.HTTP, localClusterName, localNodeName,
```

Comment:
Good spot. I have updated the latency values in test data so that all metrics are having different value, hopefully the test will help to pick up this issue.